### ISSUE TITLE
Multiline syntax that is normally invalid is accepted inside stringized annotations.

### ISSUE THREAD

**Issue by elenakrittik** [microsoft/pyright#6940]
**Describe the bug**
Multiline syntax that is normally invalid is accepted inside stringized annotations.

**Code or Screenshots**
If possible, provide a minimal, self-contained code sample (surrounded by triple back ticks) to demonstrate the issue. The code should define or import all referenced symbols.

```python
from typing_extensions import TypeAlias


Alias1 = (
    int
    | str
    | bool
)  # parentheses required

Alias2: TypeAlias = """
    int
    | str
    | bool
"""  # mypy: Invalid type comment or annotation  [valid-type]
     # ruff (a linter): [F722] Syntax error in forward annotation
     # pyright: 0 errors, 0 warnings, 0 informations

```

**VS Code extension or command-line**
Command line, 1.1.345

**Comment by erictraut** [comment]
Hmm, that was a conscious decision I made when writing pyright's parser. The meaning seems unambiguous to me if you've enclosed the type expression in quotes. Pyright treats a triple-quoted type expression as though it's implicitly parenthesized. 

Is there a reason you want this to generate an error? I can understand why mypy would generate an error here, since it's using Python's built-in parser, which is pretty limited and rigid.

If there's a good reason for rejecting this, I'm willing to make the change in pyright. I'm not yet convinced there's a good justification, however.

**Comment by erictraut** [comment]
Here's the discussion topic if you want to follow along: https://discuss.python.org/t/newlines-within-triple-quoted-type-expressions/42833.

**Comment by erictraut** [comment]
> As a library author one of my big concerns is supporting all conformant type checkers

We're making good strides toward improving conformance across type checkers. There is now an official (although not yet comprehensive) [typing spec](https://typing.readthedocs.io/en/latest/spec/index.html), and I've been contributing to a new [conformance test](https://github.com/python/typing/blob/main/conformance/README.md). You can see the latest conformance results for the four major type checkers [here](https://htmlpreview.github.io/?https://github.com/python/typing/blob/main/conformance/results/results.html). If there are other aspects of the typing spec that you think should be standardized, please post your thoughts to the [python/typing forum](https://discuss.python.org/c/typing/32).

**Comment by elenakrittik** [comment]
Right, thank you. I didn't mean that "supporting all conformant type checkers" is challenging for me, it's only something that i keep eyes on when encountering "interesting" cases like this, sorry if i miscommunicated. You and others do a great job on keeping typings consistent!

**Comment by erictraut** [comment]
It sounds like the consensus is that pyright's approach here is preferable. A [PR has been submitted to mypy](https://github.com/python/mypy/pull/16765) so its behavior matches. I'll try to get this added to the official typing spec and conformance test suite as well. 

I'm going to close this ticket since there's no required change to pyright in this case.

**Comment by erictraut** [comment]
@charliermarsh, you may also be interested in this thread, as it sounds like ruff considers this a syntax error.

**Comment by charliermarsh** [comment]
(Thanks @erictraut! We’ll update our parser.)
